src/sentry/api/endpoints/user_details.py

```diff
@@ -60,16 +57,73 @@ class Meta:
         model = User
         # no idea wtf is up with django rest framework, but we need is_active
         # and isActive
-        fields = ('name', 'username', 'isActive', 'email')
+        fields = ('name', 'isActive')
         # write_only_fields = ('password',)
 
 
+class UserSudoSerializer(BaseUserSerializer):
+    class Meta:
+        model = User
+        fields = ('password', 'username', 'email')
+
+    def validate_password(self, attrs, source):
+        # this will raise a ValidationError if password is invalid
+        password_validation.validate_password(attrs[source])
+
+        return attrs
+
+
 class UserDetailsEndpoint(UserEndpoint):
     def get(self, request, user):
         data = serialize(user, request.user, DetailedUserSerializer())
         return Response(data)
 
+    @sudo_required
+    def protected_put(self, request, user):
+        serializer = UserSudoSerializer(user, data=request.DATA, partial=True)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        result = serializer.object
+
+        if request.DATA.get('password'):
+            if request.DATA.get('password') != request.DATA.get('passwordVerify'):
+                return Response({"detail": "New password does not match verified password"},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            # TODO(billy): Why is `user.has_usable_password()` false here
```

Comment:
usually that's if a user has never had a password set. are these sso users or something?